user/service/client/client.go

```diff
@@ -202,6 +204,23 @@ func (c *Client) Delete(ctx context.Context, id string, deleet *user.Delete, con
 			logger.WithError(err).Error("Unable to destroy profile")
 		}
 	}
+	//Sending delete user event to kafka
+	userName := *result.Username
+	logger.Infof("preparing to send delete user kafka message for %v", userName)
+	if !strings.HasSuffix(userName, "@tidepool.io") && !strings.HasSuffix(userName, "@tidepool.org") {
+		var role []string
+		if result != nil && result.Roles != nil {
+			role = *result.Roles
+		}
+		eventType := "delete-user"
+		userEventMessage := map[string]interface{}{
```

Comment:
I'd publish the entire user object.

Reply:
Background: this is what I originally was doing and from memory @derrickburns correct me if I am wrong, was advised to just just send the user ID and check the database for the state of the user. Then ran into issue with finding the user (now obviously) in the database because the user was deleted. Since marketo needed these fields I added them to the payload.

Let me know if I should just pass the entire object (dont see why not) or a further discussion is needed

Reply:
You should not read the user from the DB - you could use the API, but imo it's better to just publish the user object on the topic.

Reply:
sorry that is what I meant